python/pylibcudf/pylibcudf/column.pyx

```diff
@@ -151,6 +154,46 @@ class _Ravelled:
         self.__cuda_array_interface__ = cai
 
 
+def _parse_array_interface(iface: dict) -> tuple:
+    """
+    Parse a CUDA or NumPy array interface dict.
+    """
+    typestr = iface["typestr"]
+    data = iface.get("data")
+    shape = iface["shape"]
+    strides = iface.get("strides")
+    return typestr, data, shape, strides
+
+
+def _validate_array_metadata(
+    typestr: str,
+    itemsize: int,
+    shape: tuple[int, ...],
+    data: None | tuple[int, bool],
+    strides: None | tuple[int, ...],
+) -> None:
+    """
+    Validate the CUDA or NumPy array interface metadata.
+    """
+    if typestr[0] == ">":
+        raise ValueError("Big-endian data is not supported")
+    if not isinstance(data, tuple) or not isinstance(data[0], int):
+        raise ValueError(
+            "Expected a data field with an integer pointer in the array interface. "
+            "Objects with data set to None or a buffer object are not supported."
+        )
+    if len(shape) > 2:
+        ValueError("Only 1D or 2D arrays are supported")
+    if not is_c_contiguous(shape, strides, itemsize):
+        raise ValueError("Data must be C-contiguous")
+    if len(shape) > 1:
+        total_rows = functools.reduce(operator.mul, shape[:-1])
+    else:
+        total_rows = shape[0]
+    if total_rows >= numeric_limits[size_type].max():
+        raise ValueError("Number of rows exceeds size_type limit for offsets column.")
```

Comment:
*suggestion*: This validation is checking that we have appropriate values for passing to `_from_gpumemoryview`. I suggest that the validation function should _return_ the pre-processed metadata (shape, strides, etc..) (raising if preconditions are not satisfied) needed for `_from_gpumemoryview`.

Reply:
I recombined the functions (validate + parse)